RHDEVDOCS-3752 Document Pipelines as code (#43308)
openshift/openshift-docs: modules/op-installing-pipelines-as-code-on-an-openshift-cluster.adoc

// This module is included in the following assembly:
//
// *cicd/pipelines/using-pipelines-as-code.adoc

:_content-type: PROCEDURE
[id="installing-pipelines-as-code-on-an-openshift-cluster_{context}"]
= Installing Pipelines as Code on an OpenShift cluster

> 

Reviewer: Please add an introductory line here that can be used as an abstract.

[discrete]
.Procedure

. To install Pipelines as Code on a {product-title} cluster, run the following command:
+
[source,terminal]
----
$ VERSION=0.5.4
$ oc apply -f https://raw.githubusercontent.com/openshift-pipelines/pipelines-as-code/release-$VERSION/release-$VERSION.yaml
----
+
[NOTE]
====
For the latest stable version, check the link:https://github.com/openshift-pipelines/pipelines-as-code/releases[release page]. In addition, check the {pipelines-title} release notes to ensure that the Pipelines as Code version is compatible with the {pipelines-shortname} version.
====
+

> Pipelines as Code is installed in the `pipelines-as-code` namespace, along with creation of roles and route URL for the Pipelines as Code controller.

Reviewer: ```suggestion
Pipelines as Code is installed in the `pipelines-as-code` namespace. As part of this installation process, user roles and the route URL for the Pipelines as Code controller are also created in the namespace.
```
This sentence seems awkward to me. Consider breaking it into two sentences as I suggest above.

. Note the route URL for the Pipelines as Code controller created on the cluster:
+
[source,terminal]
----
$ echo https://$(oc get route -n pipelines-as-code el-pipelines-as-code-interceptor -o jsonpath='{.spec.host}')
----
+

> This URL is referenced while configuring the Git repository hosting service provider.

Reviewer: ```suggestion
This URL will be needed later when you configure the Git repository hosting service provider.
```

> . (Optional) To allow non-administrative users to create repository custom resource definitions (CRD) in their respective namespaces, create a `RoleBinding` object of name `openshift-pipeline-as-code-clusterrole` on the namespace. For example, to allow an user to create a repository CRD in the `user-ci` namespace, run the following command:

Reviewer: ```suggestion
. (Optional) To allow non-administrative users to create repository custom resource definitions (CRDs) in their respective namespaces, create a `RoleBinding` object of name `openshift-pipeline-as-code-clusterrole` on the namespace. For example, to allow an user to create a repository CRD in the `user-ci` namespace, run the following command:
```

// This module is included in the following assembly:
//
// *cicd/pipelines/using-pipelines-as-code.adoc

:_content-type: PROCEDURE
[id="installing-pipelines-as-code-on-an-openshift-cluster_{context}"]
= Installing {pac} on an {product-title}

[role="_abstract"]
Starting with {pipelines-title} 1.7, {pac} is installed by default when you install the {pipelines-title} Operator. If you are using {pipelines-shortname} 1.7 or later versions, skip the procedure for manual installation of {pac}.

> However, if you want to disable the default installation of {pac} with the {pipelines-title} Operator, set the value of the `enablePipelinesAsCode` field as `false`k in the `TektonConfig` custom resource.

Reviewer 2: ```suggestion
However, if you want to disable the default installation of {pac} with the {pipelines-title} Operator, set the value of the `enablePipelinesAsCode` field as `false` in the `TektonConfig` custom resource.
```

[source,yaml]
----
...
spec:
    addon:
      enablePipelinesAsCode: false
...
----

[discrete]
.Procedure

> . To manually install {pac} on a {product-title} cluster instead of the default installation with the {pipelines-title} Operator, run the following command:

Reviewer 2: do we need to keep manual install now? as it is possible through operator?
cc @chmouel

Reviewer 2: if user change `enablePipelinesAsCode` to false and later decide to install it again, then they just need to change it back to true
